pkg/reconciler/ingress/ingress.go

```diff
@@ -77,6 +81,18 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ing *v1alpha1.Ingress) r
 		}
 		if ready {
 			external, internal := config.ServiceHostnames()
+
+			ns, err := r.kubernetesClient.CoreV1().Namespaces().Get(ctx, ing.Namespace, metav1.GetOptions{})
```

Comment:
Is this going to go through the informer cache this way? I'd expect to see a `NamespacesLister` injected here.

Reply:
IIRC this is a direct call. I originally wrote the code to use a `NamespacesLister/Informer` in order to watch for namespace changes but that's not needed. The advantage of a direct call is that there is no need for a cache (saving a tiny bit of memory), as the expense of additional http requests (but no watch).

I'm happy to revert to back to using `NamespacesLister` if you feel it's a better compromise.

Reply:
see my comment below. I will use an informer.